accepted/future-releases/abstract-external-fields/implementation-plan.md

```diff
@@ -0,0 +1,157 @@
+
+This feature is only enabled when null-safety is also enabled. A tool
+should emit an error message and stop if an attempt is made to enable
+this feature without null-safety.
+
+The experiment flag will be required during the period where the
+feature has not yet been released, so it may be implemented incompletely
+and there may be breaking changes to the static and dynamic properties
+of the feature until the release. When released, the feature will be
+enabled without the flag, and support for the flag will be removed
+in later stable releases.
+
+### Tests
+
+The language team adds
+[tests](https://github.com/dart-lang/sdk/tree/master/tests/language/external_abstract_fields)
+for the feature. The Co19 team adds tests for the specification.
+
+## Phase 1 (Front End Implementation)
+
+### CFE
+
+The CFE implements parsing the new syntax, detecting static errors, and
+performing the transformations that turns the new variable declarations into
+external or abstract getters and/or setters.
+
+### Analyzer
+
+The analyzer implements parsing the new syntax and checking for static
+errors. Based on input from Paul Berry, this is unlikely to rely on
+desugaring, and approximately the following changes will be needed in the
+analyzer:
+
+- Modify the AST and summary representations to include support for
+  `abstract` and `external` keywords on variables.
+- Modify the mechanism for building element models to mark abstract fields
+  as abstract, and external fields as external.
+- Modify the error generation logic to suppress 'uninitialized field'
+  errors for abstract and external fields.
+- Modify the error generation logic to report errors if an abstract or
+  external field is initialized.
+- There may be a need for changes to ensure that an appropriate error is
+  reported if a concrete class fails to implement a getter or setter which
+  was introduced by an abstract variable.
+- There may be a need to change the constant evaluator to avoid computing
+  a value for an abstract/external variable.
+
+## Phase 2 (Remaining Implementation)
+
+### dart2js, DDC, VM
+
+The feature is expected to to be desugared, which means that it will not
+exist in kernel code, and backends need not implement anything.
+
+### Dartfmt
+
+Define and implement formatting rules for the new syntax.
+Add formatting tests.
+
+### Formal specification
+
+Add a formal specification of this feature to the language specification.
+
+### Cider
+
+Validate Cider support.
+
+### IntelliJ
+
+Update the IntelliJ parser to support the new syntax.
+
+### Grok
+
+Update Grok to handle the new AST.
+
+### Analysis server
+
+No additional work is expected for the analysis server.
+
+### Atom plug-in
+
+The [Dart Atom plug-in][atom] has a grammar for syntax highlighting
+Dart code in Atom. This same grammar is also used for syntax highlighting
+on GitHub. Update this to handle the new syntax.
+
+[atom]: https://github.com/dart-atom/dart
+
+### VS Code
```

Comment:
This section can be removed (or, re-worded to reference the above section).

Reply:
Reworded as suggested.